### DISCUSSION TITLE
Anvil GUI

### DISCUSSION THREAD

**Discussion by ghost** [Minestom/Minestom#833] [Help]
Hi. How can I get text from input in Anvil GUI?

**Comment by MrlnHi** [comment]
You have to listen for the ClientNameItemPacket using
```java
MinecraftServer.getPacketListenerManager().setListener()
```
it gets sent everytime the player changes the text input, so you can save it onto them using a Tag for example

**Reply by ghost** [reply]
Can you, please, provide an example?

**Reply by MrlnHi** [reply]
```java
MinecraftServer.getPacketListenerManager().setListener(ClientNameItemPacket.class, (packet, player) -> {
    System.out.println(player.getUsername() + " typed " + packet.itemName());
});
```

**Reply by ghost** [reply]
I have this code and nothing being printed out:
```java
final Inventory inventory = new Inventory(InventoryType.ANVIL, Component.text("EventName", NamedTextColor.DARK_AQUA));

MinecraftServer.getPacketListenerManager().setListener(ClientNameItemPacket.class, (packet, player2) -> {
   System.out.println(player2.getUsername() + " typed " + packet.itemName());
});

player.openInventory(inventory);

**Reply by TheMode** [reply]
Be aware that you are setting the listener, not adding one. Meaning that this will only work once, and will override what minestom has.

**Reply by ghost** [reply]
> Be aware that you are setting the listener, not adding one. Meaning that this will only work once, and will override what minestom has.

Code, that you suggested [here](https://github.com/Minestom/Minestom/discussions/833#discussioncomment-2471155) isn't working too, btw. It prints nothing

**Comment by TheMode** [comment]
```java
node.addListener(PlayerPacketEvent.class, event -> {
    if (event.getPacket() instanceof ClientNameItemPacket packet) {
        System.out.println("text " + packet.itemName());
    }
});
```
